pkg/controller/jenkins/configuration/base/resources/pod.go

```diff
@@ -204,7 +209,11 @@ func GetJenkinsMasterContainerBaseVolumeMounts(jenkins *v1alpha2.Jenkins) []core
 // NewJenkinsMasterContainer returns Jenkins master Kubernetes container
 func NewJenkinsMasterContainer(jenkins *v1alpha2.Jenkins) corev1.Container {
 	jenkinsContainer := jenkins.Spec.Master.Containers[0]
-	envs := GetJenkinsMasterContainerBaseEnvs(jenkins)
+	jenkinsHomeEnvVar := corev1.EnvVar{
+		Name:  "JENKINS_HOME",
+		Value: getJenkinsHomePath(jenkins),
+	}
+	envs := append(GetJenkinsMasterContainerBaseEnvs(jenkins), jenkinsHomeEnvVar)
 	envs = append(envs, jenkinsContainer.Env...)
```

Comment:
`jenkinsHomeEnvVar` should be added only if it is not set in `jenkinsContainer.Env`